File: programmers/level1/017.py
# 17. 나누어 떨어지는 숫자 배열
def solution(arr, divisor):
    answer = []
    for i in arr:
        if i % divisor==0:
            answer.append(i)
    answer.sort()
    if len(answer)==0:
        answer.append(-1)
    return answer


# for pythonic
def solution(arr, divisor):
    answer = sorted([ x for x in arr if x%divisor==0])
    if not answer:
        return [-1]
    return answer

File: programmers/level2/017.java
// 17. H-Index
import java.util.*;
class Solution {
    public int solution(int[] citations) {
        if(citations.length==1)
            return citations[0];
        
        Arrays.sort(citations);
        final int MAX = citations[citations.length-1];
        
        
        int[] nums = new int [MAX+1];
        for(int i = citations.length-1; i>=0; i--){
            nums[citations[i]]+=1;
        }
        
        for(int i=MAX; i>0; i--){
            if(i==MAX)
                continue;
            nums[i]=nums[i]+nums[i+1];
        }
        int answer = 0;
        for(int i=1; i<nums.length; i++){
            if(nums[i]>=i){
                answer = i;     
            }
        }
        return answer;
    }
}